zcash_primitives/src/transaction/mod.rs

```diff
@@ -54,6 +54,12 @@ pub struct Transaction {
     data: TransactionData,
 }
 
+impl Transaction {
```

Comment:
The problem with `TransactionData::freeze` is that I can't specify a fake TxID. What I want (for my tests) is to create a Fake Transaction with a pre-determined TxID, so that I can test some Transparent Address beheaviour in Zecwallet.

Reply:
Can you possibly give us a link to the test?

Reply:
https://github.com/adityapk00/zecwallet-light-cli/blob/master/lib/src/lightwallet/tests.rs#L248

Used here: 
https://github.com/adityapk00/zecwallet-light-cli/blob/master/lib/src/lightwallet/tests.rs#L374

Reply:
A better way to go here would be to construct a fake compact block from a fake transaction - and it looks like this is something that we could use within our code as well. @str4d and I will pair tomorrow on making up some generalized test utilities that can go into zcash_client_backend to help share a standard approach to this kind of compact block & transaction construction.

Reply:
Perfect. Thanks a ton!